test/e2e/framework/util.go

```diff
@@ -275,6 +240,26 @@ func GetFreePort(t *testing.T) (string, error) {
 	}
 }
 
+// GetMaybeFreePort returns a port that was free right before being
+// returned to the caller. This is good enough for interactive use
+// where process start could be trivially retried, but GetFreePort
+// should be preferred for parallel use cases where there may be
+// contention for port allocation.
+func GetMaybeFreePort() (string, error) {
```

Comment:
I'm using this in the invoker for a shared test server (one command for server consistent with test expectations for development and ci). There is no `t` in this context, as required by `GetMaybeFreePort` for logging and for test dir creation/post test cleanup. It was cheaper for me to rewrite as this than to update `GetFreePort` to work in both contexts. If you'd like to propose a single solution that works for both cases, by all means, but for me it wasn't worth the effort.

Reply:
`GetFreePort()` is pretty small. A refactor to an interface instead of a `*testing.T` seems relatively straightforward. Didn't @sttts or someone add a flag to `kcp` to just use some port and resolve it in-process, then write it to a file? Why not just use that?